data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/JiraService.java

```diff
@@ -0,0 +1,360 @@
+        log.trace("Creating item information and adding in queue");
+        jiraProjectCache.keySet().forEach(key -> {
+            Map<String, Object> metadata = new HashMap<>();
+            metadata.put(CONTENT_TYPE, JiraContentType.PROJECT.getType());
+            ItemInfo itemInfo = createItemInfo(_PROJECT + key, metadata);
+            itemInfoQueue.add(itemInfo);
+        });
+
+    }
+
+    /**
+     * Method for building Issue Item Info.
+     *
+     * @param configuration Input Parameter
+     * @param timestamp     Input Parameter
+     */
+    private void searchForNewTicketsAndAddToQueue(JiraSourceConfig configuration, Instant timestamp,
+                                                  Queue<ItemInfo> itemInfoQueue, List<Future<Boolean>> futureList,
+                                                  ExecutorService crawlerTaskExecutor) {
+        log.trace("Looking for Add/Modified tickets with a Search API call");
+        StringBuilder jql = createIssueFilterCriteria(configuration, timestamp);
+        int total;
+        int startAt = 0;
+        try {
+            do {
+                SearchResults searchIssues = getAllIssues(jql, startAt, configuration);
+                List<IssueBean> issueList = new ArrayList<>(searchIssues.getIssues());
+                total = searchIssues.getTotal();
+                startAt += searchIssues.getIssues().size();
+                futureList.add(crawlerTaskExecutor.submit(
+                        () -> addItemsToQueue(issueList, itemInfoQueue), false));
+            } while (startAt < total);
+            searchResultsFoundCounter.increment(total);
+            log.info("Number of tickets found in search api call: {}", total);
+        } catch (RuntimeException ex) {
+            log.error("An exception has occurred while fetching"
+                    + " issue entity information , Error: {}", ex.getMessage());
+            throw new BadRequestException(ex.getMessage(), ex);
+        }
+    }
+
+    /**
+     * Add items to queue.
+     *
+     * @param issueList     Issue list.
+     * @param itemInfoQueue Item info queue.
+     */
+    private void addItemsToQueue(List<IssueBean> issueList, Queue<ItemInfo> itemInfoQueue) {
+        issueList.forEach(issue -> {
+            Map<String, Object> issueMetadata = new HashMap<>();
+            if (Objects.nonNull(((Map) issue.getFields().get(PROJECT)).get(KEY))) {
+                issueMetadata.put(PROJECT_KEY,
+                        ((Map) issue.getFields().get(PROJECT)).get(KEY).toString());
+            }
+            if (Objects.nonNull(((Map) issue.getFields().get(PROJECT)).get(NAME))) {
+                issueMetadata.put(PROJECT_NAME,
+                        ((Map) issue.getFields().get(PROJECT)).get(NAME).toString());
+            }
+
+            long created = 0;
+            if (Objects.nonNull(issue.getFields()) && issue.getFields().get(CREATED)
```

Comment:
This comparison with 23 and converting to timestamp looks hacky. Is there no better way? Like trying to parse string to date using existing Date/Time parse formats?